### ISSUE TITLE
use-ember-get-and-set and model relationships test

### ISSUE THREAD

**Issue by netes** [netguru/eslint-plugin-ember#28]
Consider this code in model unit test:
```js
test('user relationship', function(assert) {
  const contact = this.store().modelFor('contact');
  const relationship = get(contact, 'relationshipsByName').get('user');
  assert.equal(relationship.key, 'user');
  assert.equal(relationship.kind, 'belongsTo');
});
```

This will throw an error because we're not using `Ember.get` getter for the `user` property. But we can't do it because `relationshipsByName` is an `Ember.Map` which has different `get` function.

My first idea was to add some optional settings to this rule where we would define particular preceding properties which shouldn't throw an error.

I'm almost sure that this won't be the only case, especially when we're using some JQuery addons which can also have their own `getters` (for example pickadate.js).

WDYT?

**Comment by Padchi** [comment]
Same thing if you using ember-cli-mirage and using get.

**Comment by michalsnik** [comment]
This might a problem, and I'm not sure how we should actually handle this. We can't predict if current object is EmberObject or not. The only solution I see for now is to add additional setting called e.g.`onlyThis` which will shift this rule to check only `this.<get|set>(` and it won't care about any other `<*>.get(` calls.

What do you think @netes ?

**Comment by netes** [comment]
Seems like a decent workaround but I'm not 100% sure we really need this. All in all, we can just disable next line check when such problem appears.

**Comment by michalsnik** [comment]
I was just thinking about it. What if we add an optional setting `disableInTests` that will simply disable this rule in all tests? Plus we'll disable this rule for `mirage` folder by default. I think it would be the easiest and most reliable solution. What do you think @netes? I'm happy to work on this.

**Comment by netes** [comment]
I'm not sure if it's a good solution because by doing this (`disableInTests`) we disallow rule in the favor of some few edge cases. Also, we can just use second `.eslintrc` in `tests` folder to disable particular rules, no need for `disableInTests` option then.

**Comment by michalsnik** [comment]
Ah true, are we closing this issue then @netes?

**Comment by netes** [comment]
I think so :)

**Comment by jonpitch** [comment]
I have a use case where I often override mirage routes in a test in order to simulate different API results. For example:

```javascript
//  /mirage/config.js
this.get('some/route', function() {
  return { status: 'ok' };
});

// /tests/acceptance/some-test.js
test('i want to verify what happens when the api throws a 500', function(assert) {
  server.get('some/route', function() {
    return new Mirage.Response(500, { }, {
      errors: [{
        code: '500',
        title: 'run for your life'
      }]
    });
  });

  // assert things
});
```

Currently `server.get` throws a linting error. Has there been any discussion about a whitelist functionality or ignoring `server.get` entirely? This is listed in the [Mirage docs](http://www.ember-cli-mirage.com/docs/v0.3.x/acceptance-testing/#asserting-a-server-call-was-made-in-a-test). My current workaround is to:

```javascript
const { get: GET } = server;
GET('some/route', ...);
```
